server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -211,9 +211,7 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
                 settings,
                 Names.SEARCH_WORKER,
                 searchOrGetThreadPoolSize,
-                // TODO the queue is only needed as overflow because we may release permits before the executor updated its internal state.
-                // Queue size equal to the bound (max pool size) should be enough to ensure no rejection, but it could also be unbounded.
-                searchOrGetThreadPoolSize,
+                -128, // TODO clean this up
```

Comment:
We already use "special" numbers to indicate an action ('-1' is one such number). Using a `-128` seems, eh?

Why not add a constructor to `FixedExecutorBuilder` that accepts a manually created queue?